### ISSUE TITLE
MultiStepLR is broken

### ISSUE THREAD

**Issue by sovrasov** [pytorch/pytorch#20486]
## 🐛 Bug

MultiStepLR drops learning rate by gamma**2 times instead of gamma on each milestone. After one epoch it restores the lr to normal value.

## To Reproduce

Use the following code:
```python
import torch
params = [torch.nn.Parameter(torch.randn(1, 1))]
optimizer = torch.optim.SGD(params, lr=0.2)
scheduler = torch.optim.lr_scheduler.MultiStepLR(optimizer, milestones=[2,8], gamma=0.1)
for epoch in range(10):
    print(scheduler.get_lr())
    scheduler.step()
```
The output is:
[0.2]
[0.2]
[0.0020000000000000005] here should be 0.02
[0.020000000000000004]
[0.020000000000000004]
[0.020000000000000004]
[0.020000000000000004]
[0.020000000000000004]
[0.00020000000000000006] here should be 0.002
[0.0020000000000000005]

## Expected behavior

In the provided sample the learning rate should be adjusted only twice.

## Environment

PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.5.0-12ubuntu1~16.04) 5.5.0 20171010
CMake version: version 3.9.1

Python version: 3.5
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce RTX 2080 Ti
GPU 1: GeForce GTX 1080 Ti
GPU 2: GeForce GTX 1080 Ti

Nvidia driver version: 418.56
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.16.3
[pip3] torch==1.1.0
[pip3] torchstat==0.0.7
[pip3] torchvision==0.2.2.post3
[conda] Could not collect

**Comment by pietern** [comment]
Adding the "needs discussion" label because I don't know if this behavior is by design or not.

I agree it looks faulty but need someone else to confirm this.

**Comment by sovrasov** [comment]
@pietern the same issue is reported here https://github.com/pytorch/pytorch/issues/20527
Behavior of the schedulers has been certainly changed since 1.0.1 with no changes in documentation at the same time.

**Comment by grib0ed0v** [comment]
@sovrasov @pietern I can confirm that I also noticed such kind of behavior.
For one of my trainings, MultiStep LR graph is attached below:
![image](https://user-images.githubusercontent.com/10911871/57837202-f8a25c80-77ca-11e9-8d59-e6915a550bc3.png)
I'm sure that this doesn't work as designed.

**Comment by ezyang** [comment]
We're going to close this as a duplicate of #20527 and yes we agree this is high priority and a bug.
